### PR TITLE
Change: Add China Propaganda Center to Prerequisite of Internet Center

### PR DESCRIPTION
**Pull request by xezon**
This change adds the Propaganda Center or Warfactory prerequisite to the Internet Center. Reason being, that Hackers also require the Propaganda Center. This makes the requirement more intuitive.

## Original

Internet Center requires Warfactory.

## Patched

Internet Center requires Warfactory OR Propaganda Center.

### CODE REVIEW

**Comment by xezon** [comment]
Could this change break any Campaign mission?

**Comment by commy2** [comment]
You could use the IC to access Sathack I without the intention of building any Hackers. Not that that is useful much, but you could.

**Comment by Jundiyy** [comment]
I dislike this change, whilst I understand the reasoning, it's not a needed a change and does ever so slightly nerf China. Of course, this is mainly in FFA as IC is hardly built in 1v1s. 2v2s/team games? Not sure how often IC is built for vision.
Think this should not be implemented.

**Comment by Stubbjax** [comment]
The primary rationale for this change is that it makes far more logical and intuitive sense that the Internet Centre's availability is connected to Hacker availability, as the two are effectively counterparts that belong together. It is incredibly rare and often disadvantageous to construct an Internet Centre _before_ a Propaganda Centre, so there is little merit to the minuscule loss of flexibility as a result of this change.

A lot of confusion arises when a player has a Propaganda Centre but has lost their War Factories, and can no longer construct an Internet Centre to protect their existing Hackers. A solution that would hopefully satisfy the purists while also solving the counterintuitive building prerequisite would be to add the Propaganda Centre as an additional (optional) prerequisite.
```
Object ChinaInternetCenter
  Prerequisites
    Object = ChinaWarFactory ChinaPropagandaCenter
  End
End
```

**Comment by xezon** [comment]
500 IQ suggestion @Stubbjax

**Comment by xezon** [comment]
Updated.

**Comment by Jundiyy** [comment]
Yes, I think that's a better change.

**Comment by ExiLe-ZH** [comment]
What about losing your Propaganda Center? Then you can't (re)build an IC to protect your Hackers?

I also think the Satellite Upgrades should be more accessible with this patch, this change doesn't help then.

Or am I understanding it all wrong?

**Comment by xezon** [comment]
In layman terms:

## Original

Internet Center requires Warfactory.

## Patched

Internet Center requires Warfactory OR Propaganda Center.

**Comment by ExiLe-ZH** [comment]
> In layman terms:
> ## Original
> 
> Internet Center requires Warfactory.
> ## Patched
> 
> Internet Center requires Warfactory OR Propaganda Center.

Ahhhh the OR is VERY PROFESSIONAL!
